### DISCUSSION TITLE
IPNS support

### DISCUSSION THREAD

**Discussion by 2color** [n0-computer/iroh#226] [Q&A] [answered]
Does Iroh support IPNS resolution and publishing?

If so, which transports are supported? DHT, PubSub or both?

**Comment by dariusc93** [comment]
not affiliated with the project so i wouldnt know what is done in the backend, but so far code wide IPNS isnt implemented.

**Comment by b5** [comment, marked as answer]
We don't yet support IPNS. We know we need something that "does the mutability thing", and having IPNS would be very helpful, but we want time to understand, assess, ideally address the performance issues that have plagued IPNS in the past before committing to supporting IPNS.

Everything that goes into iroh has to just work. We can't ship IPNS until we understand how to make IPNS robust.

**Reply by dignifiedquire** [reply]
See #482 for tracking
